packages/lib/server/repository/workflow-permissions.ts

```diff
@@ -0,0 +1,169 @@
+import type { Workflow } from "@prisma/client";
+
+import { isAuthorized } from "@calcom/trpc/server/routers/viewer/workflows/util";
+
+export interface WorkflowPermissions {
+  canView: boolean;
+  canUpdate: boolean;
+  canDelete: boolean;
+  canManage: boolean;
+  readOnly: boolean; // Keep for backward compatibility
+}
+
+interface TeamPermissionsCache {
+  [teamId: string]: WorkflowPermissions;
```

Comment:
Not a true cache. Just for this RUN of fetching workflows. we cache by teamId what permissions the user has access too instead of querying every time